pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -89,6 +90,35 @@ func (pl *VolumeBinding) Name() string {
 	return Name
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEvent {
+	events := []framework.ClusterEvent{
+		// Pods may fail because the storage class is not configured correctly.
+		// After the storage class is fixed (e.g. allowedTopologies,
+		// volumeBindingMode), the pods may become schedulable.
+		{Resource: framework.StorageClass, ActionType: framework.Add | framework.Update},
+		// We binds PVCs with PVs, so any changes may make the pods schedulable.
+		{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update},
+		{Resource: framework.PersistentVolume, ActionType: framework.Add | framework.Update},
+		// Pods may fail to find available PVs because the node labels does not
+		// fix in the storage class's allowed topoligies or PV's node affinity
+		// criteria. Changes to node may make pods schedulable.
+		{Resource: framework.Node, ActionType: framework.Add | framework.Update},
+		// We relies on CSI node to translate in-tree PV to CSI.
+		{Resource: framework.CSINode, ActionType: framework.Add | framework.Update},
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity) {
```

Comment:
we are trying to cut the dependency on feature gates, please follow up with a PR to remove this, see as an example inter-pod affinity https://github.com/kubernetes/kubernetes/blob/23d4b3b4f0d8b844f01b93262737c3d0efb6bff7/pkg/scheduler/framework/plugins/interpodaffinity/plugin.go#L58

Reply:
Yes. Something to add: volumebinding used to depend on featuregate already, it's not a new dependency here.

Reply:
IIUC, it's ok to use `utilfeature.DefaultFeatureGate.Enabled(...)` now, but it's deprecated and must be migrated to  "k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"`?

I'll create a follow-up PR later.

Reply:
> do you have a PR ready for this feature gate refactoring? I just don't want this follow up to get lost.

not yet, created an issue for this https://github.com/kubernetes/kubernetes/issues/103431